firebase-firestore/src/main/java/com/google/firebase/firestore/local/TargetData.java

```diff
@@ -170,6 +200,7 @@ public int hashCode() {
     result = 31 * result + snapshotVersion.hashCode();
     result = 31 * result + lastLimboFreeSnapshotVersion.hashCode();
     result = 31 * result + resumeToken.hashCode();
+    result = 31 * result + (expectedCount != null ? expectedCount.hashCode() : 0);
```

Comment:
To correctly handle `null` when calculating the hash code, you can use `Objects.hashCode()`. Please do that here because it is a bit more readable.

https://developer.android.com/reference/java/util/Objects?hl=en#hashCode(java.lang.Object)